### PR TITLE
test(repo): setup unit tests

### PR DESCRIPTION
**Pull request by QZera**
Closes [a4c047](https://app.radicle.xyz/nodes/seed.radicle.garden/rad:z2BdUVZFvHdxRfdtGJQdSH2kyXNM6/issues/a4c047117652a8c9cbeacca6532e90fff7fd8229)

### CODE REVIEW

File: test/components/ColumnCard.nuxt.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        },
+      },
+      href: 'https://example.com',
+    }
+    await renderSuspended(ColumnCard, {
+      props,
+    })
+
+    expect(screen.getByText(props.status.name)).toBeInTheDocument()
+    expect(screen.getByText(props.id.slice(0, 7))).toBeInTheDocument()
```
Comment (gsaslis) [inline test/components/ColumnCard.nuxt.test.ts:25]: Can this `7` be a constant (with a name that conveys the meaning behind "7")  ? ;)

Comment (QZera) [reply test/components/ColumnCard.nuxt.test.ts]: Added the `taskTruncatedIdLength` constant

File: eslint.config.js [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    ignores: ['*.md'],
+    typescript: {
+      tsconfigPath: 'tsconfig.json',
+    },
   },
-})
+  {
+    files: ['**/*.test.ts'],
+    rules: {
+      'max-len': ['error', 120],
```
Comment (maninak) [inline eslint.config.js:13]: Why do we need this?

File: test/components/ColumnCard.nuxt.test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { screen } from '@testing-library/vue'
+import { renderSuspended } from '@nuxt/test-utils/runtime'
+import { ColumnCard } from '#components'
+import { taskTruncatedIdLength } from '~/constants/tasks'
+
+describe('<ColumnCard />', () => {
+  it('should render the title (link to a new tab), status, id (truncated), and labels', async () => {
```
Comment (maninak) [inline test/components/ColumnCard.nuxt.test.ts:7]: Would be good if we would cover edge cases for this component too. For example every time there's a branching e.g. this [`v-if`](https://github.com/cytechmobile/radicle-planning-boards/pull/73/files#diff-4bdcf814fcd7c205d380617026790b5055c0e9a6a7dc4f49973cb0fe62b00728R41)

But for this specific case, I don't see it as a hard requirement. Up to you.

**Review by maninak — COMMENTED** [review]
Setup looks good, well done!

I only found minor things.

Also it would be good to add a reference to the awesome [vitest VS Code extension](https://marketplace.visualstudio.com/items?itemName=vitest.explorer) among our [recommended extensions](https://github.com/cytechmobile/radicle-planning-boards/blob/main/.vscode/extensions.json#L4).

**Comment by maninak** [comment]
The github action issue that seems to be unrelated to the changes of this PR may be resolved with https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#r1636317563

File: eslint.config.js (hunk 2 above)
Comment (QZera) [reply eslint.config.js]: The name of the test I wrote was triggering the 95 default character limit.

Looked into the rule a bit more, and found out there's an `ignoreStrings` option, which covers this case.

There's also the `ignorePattern` option, which would let us be more specific (e.g. ignore lines starting with `it(`) but I feel `ignoreStrings` is good enough.

Unfortunately, I needed to re-iterate the whole rule and couldn't just update the `ignoreStrings` option.

File: test/components/ColumnCard.nuxt.test.ts (hunk 3 above)
Comment (QZera) [reply test/components/ColumnCard.nuxt.test.ts:7]: I agree that edge cases should be covered by tests. But in this specific case I don't think it would add any value.

It would be the same as testing that the title doesn't render when an empty string is passed. Pretty redundant.

**Comment by QZera** [comment]
> Setup looks good, well done!
> 
> I only found minor things.
> 
> Also it would be good to add a reference to the awesome [vitest VS Code extension](https://marketplace.visualstudio.com/items?itemName=vitest.explorer) among our [recommended extensions](https://github.com/cytechmobile/radicle-planning-boards/blob/main/.vscode/extensions.json#L4).

Good point. Added it.

File: test/components/ColumnCard.nuxt.test.ts (hunk 3 above)
Comment (maninak) [inline test/components/ColumnCard.nuxt.test.ts:7]: Just saw this. If we're using the `it()` assertion, then the test description shouldn't use the "should" word in it. That's an alternative notation `should()`. Either way, the string inside those should be in the present tense and as concise as possible (think formal spec/design document). In this case, I'd personally go with:

```suggestion
  it('renders the title (link as new tab), status, id (truncated), and labels', async () => {
```

The above also negates the [need for the linting rule change](https://github.com/cytechmobile/radicle-planning-boards/pull/73#discussion_r1636204954) to begin with. At least for this one case, but it rules are there for a reason. If the description of a test case is too long it may be able to be worded more concisely, it may be doing too much and if all else fail then we should shamefully `// eslint-disable-next-line max-len`.

File: test/components/ColumnCard.nuxt.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        icon: {
+          name: 'octicon:issue-opened-16',
+          class: '',
+        },
+      },
+      href: 'https://example.com',
+    }
+    await renderSuspended(ColumnCard, {
+      props,
+    })
```
Comment (maninak) [inline test/components/ColumnCard.nuxt.test.ts:23]: nit: new line for separation between "chapters": setup, execution, finalization. Also, consolidate the execution part into a single line because we can.

```suggestion

    await renderSuspended(ColumnCard, { props })
```

**Review by maninak — CHANGES_REQUESTED** [review]

File: eslint.config.js (hunk 2 above)
Comment (maninak) [reply eslint.config.js]: I haven't used the ignoreStrings on purpose. Too long strings in the source code should be broken down into multiple lines either with a `"" + ""` or as template strings

```js
const str = `multi\
line\
string`
```

In the case of tests I wouldn't mind too much if they were _sometimes_ longer e.g. 120 columns, but to be honest, since the code there is max 95 cols, it would still stand out and impede readability in narrower viewports.

Most importantly see [this inline comment](https://github.com/cytechmobile/radicle-planning-boards/pull/73#discussion_r1641789857) for more argumentation as to why I'm resisting this change (but I'm open to revisiting this decision later).

File: test/components/ColumnCard.nuxt.test.ts (hunk 4 above)
Comment (QZera) [reply test/components/ColumnCard.nuxt.test.ts]: I consider rendering the component part of the "setup" or "arrange" chapter. This is also how it is treated in [this example](https://testing-library.com/docs/react-testing-library/example-intro#quickstart) from the `testing-library` docs.

I guess you could count it as "execution" or "act" in this case since there's no user action.

Updated as suggested.

File: eslint.config.js (hunk 2 above)
Comment (QZera) [reply eslint.config.js]: Sounds good, reverted the eslint changes.

File: test/components/ColumnCard.nuxt.test.ts (hunk 3 above)
Comment (QZera) [reply test/components/ColumnCard.nuxt.test.ts]: 👍 Updated as suggested

File: test/components/ColumnCard.nuxt.test.ts (hunk 4 above)
Comment (maninak) [reply test/components/ColumnCard.nuxt.test.ts]: I think you're more right than me here. Since this is just a minor style nit I'll approve this version now, but feel free to revert it in any later patch. What matters most to me is alignment and knowing we have a systematic approach, so as far as I'm concerned that much was achieved here. :+1:

**Review by maninak — APPROVED** [review]
Yay, we gots testing, well done!
